incubator/hnc/internal/pkg/selectors/selectors.go

```diff
@@ -97,3 +103,24 @@ func getSelectorFromString(str string) (labels.Selector, error) {
 	}
 	return selector, nil
 }
+
+// GetNoneSelector returns true indicates that user do not want this object to be propagated
+func GetNoneSelector(inst *unstructured.Unstructured) (bool, error) {
+	noneSelectorStr := GetNoneSelectorAnnotation(inst)
+	// Empty string is treated as 'false'. In other selector cases, the empty string is auto converted to
+	// a selector that matches everything.
+	if noneSelectorStr == "" {
+		return false, nil
+	}
+	noneSelector, err := strconv.ParseBool(noneSelectorStr)
+	if err != nil {
+		// When the user put an invalid noneSelector, we choose not to propagate this object to any child
+		// namespace to protect any object in the child namespaces to be overwritten
+		return true,
+			fmt.Errorf("invalid %q value, it should be either true, false or empty, but got %q",
```

Comment:
Why are we generating our own error and not returning a wrapped version of the one from `ParseBool`?